src/main/java/org/junitpioneer/jupiter/json/AbstractJsonArgumentsProvider.java

```diff
@@ -55,15 +55,14 @@ private static Object createArgumentForCartesianProvider(Parameter parameter, No
 	}
 
 	private static Arguments createArguments(Method method, Node node) {
-		boolean singleParameter = method.getParameterCount() == 1;
-		if (singleParameter) {
+		if (method.getParameterCount() == 1) {
 			Parameter onlyParameter = method.getParameters()[0];
 			// When there is a single parameter, the user might want to extract a single value or an entire type.
 			// When the parameter has the `@Property` annotation, then a single value needs to be extracted.
 			Property property = onlyParameter.getAnnotation(Property.class);
 			if (property == null) {
 				// no property specified -> the node should be converted in the parameter type
-				return Arguments.arguments(node.toType(onlyParameter.getType()));
+				return () -> new Object[] { node.toType(onlyParameter.getParameterizedType()) };
```

Comment:
What is wrong with using `Arguments.arguments`?

Reply:
This reminded me of assertj/assertj@6cab52be094ece12818cc41f82028ccf833af201 and junit-team/junit5#2708, not sure if they are related to this one.

Reply:
It's a weird generics issue as far as I understood - we call a method that returns a `T`, pass it to a method with varargs and it gets interpreted as an array. But we return a single object, not an array and it throws a `ClassCastException`.

Reply:
It's fairly trivial to reproduce the problem:
```java
public static void main(String[] args) {
	methodWithVarargs(getGenericThing());
}

public static void methodWithVarargs(Object... arguments) {
	System.out.println(Arrays.toString(arguments));
}

public static <T> T getGenericThing() {
	return (T) new Object();
}
```

This throws:
```
Exception in thread "main" java.lang.ClassCastException: class java.lang.Object cannot be cast to class [Ljava.lang.Object; (java.lang.Object and [Ljava.lang.Object; are in module java.base of loader 'bootstrap')
```

Reply:
I will add a comment explaining it to the code, to avoid future confusion.

Reply:
Got it. If you cat to `Object` you should not have the problem. In any case, thanks for the explanation

Reply:
I'm actually not sure about this line - would it look nicer with an `(Object)` cast?